src/miniscript/context.rs

```diff
@@ -71,6 +80,11 @@ pub trait ScriptContext:
     fn check_frag_validity<Pk: MiniscriptKey, Ctx: ScriptContext>(
         _frag: &Terminal<Pk, Ctx>,
     ) -> Result<(), ScriptContextError>;
+
+    fn sighash(
+        psbt: &psbt::PartiallySignedTransaction,
+        input_index: usize,
+    ) -> Result<(SigHash, SigHashType), SignerError>;
```

Comment:
This is a cool idea, putting this into `ScriptContext`. I think it should be changed though as

```
fn sighash(tx: &bitcoin::Transaction, input_index: usize, sighash_type: SigHashType) -> SigHash;
```

Then the function is more generic than just PSBT, and also it can't fail.

Reply:
The existing functionality I think is still worth including in the library, but as a standalone function which takes a PSBT and input index, does the appropriate lookups (which may fail) then calls `ScriptContext::sighash` to do the actual sighash computation.